src/uu/dd/src/dd_unit_tests/block_unblock_tests.rs

```diff
@@ -0,0 +1,357 @@
+                                sync: None,
+                                noerror: false,
+                            },
+                            iflags: DEFAULT_IFLAGS,
+                        },
+                        Output {
+                            dst: File::create(format!("./test-resources/FAILED-{}.test", $test_name)).unwrap(),
+                            obs: 512,
+                            cflags: DEFAULT_CFO,
+                        },
+                        $spec,
+                        format!("./test-resources/FAILED-{}.test", $test_name)
+        );
+    };
+);
+
+macro_rules! make_unblock_test (
+    ( $test_id:ident, $test_name:expr, $src:expr, $unblock:expr, $spec:expr ) =>
+    {
+        make_spec_test!($test_id,
+                        $test_name,
+                        Input {
+                            src: $src,
+                            non_ascii: false,
+                            ibs: 512,
+                            xfer_stats: None,
+                            count: None,
+                            cflags: IConvFlags {
+                                ctable: None,
+                                block: None,
+                                unblock: $unblock,
+                                swab: false,
+                                sync: None,
+                                noerror: false,
+                            },
+                            iflags: DEFAULT_IFLAGS,
+                        },
+                        Output {
+                            dst: File::create(format!("./test-resources/FAILED-{}.test", $test_name)).unwrap(),
+                            obs: 512,
+                            cflags: DEFAULT_CFO,
+                        },
+                        $spec,
+                        format!("./test-resources/FAILED-{}.test", $test_name)
+        );
+    };
+);
+
+#[test]
+fn block_test_no_nl() {
+    let mut rs = rs!();
+    let buf = vec![0u8, 1u8, 2u8, 3u8];
+    let res = block(buf, 4, &mut rs);
+
+    assert_eq!(res, vec![vec![0u8, 1u8, 2u8, 3u8],]);
+}
+
+#[test]
+fn block_test_no_nl_short_record() {
+    let mut rs = rs!();
+    let buf = vec![0u8, 1u8, 2u8, 3u8];
+    let res = block(buf, 8, &mut rs);
+
+    assert_eq!(
+        res,
+        vec![vec![0u8, 1u8, 2u8, 3u8, SPACE, SPACE, SPACE, SPACE],]
+    );
+}
+
+#[test]
+fn block_test_no_nl_trunc() {
+    let mut rs = rs!();
+    let buf = vec![0u8, 1u8, 2u8, 3u8, 4u8];
+    let res = block(buf, 4, &mut rs);
+
+    assert_eq!(res, vec![vec![0u8, 1u8, 2u8, 3u8 /*, 4u8*/],]);
+    assert_eq!(rs.records_truncated, 1);
+}
+
+#[test]
+fn block_test_nl_gt_cbs_trunc() {
+    let mut rs = rs!();
+    let buf = vec![
+        0u8, 1u8, 2u8, 3u8, 4u8, NL, 0u8, 1u8, 2u8, 3u8, 4u8, NL, 5u8, 6u8, 7u8, 8u8,
+    ];
+    let res = block(buf, 4, &mut rs);
+
+    assert_eq!(
+        res,
+        vec![
+            vec![0u8, 1u8, 2u8, 3u8],
+            // vec![4u8, SPACE, SPACE, SPACE],
```

Comment:
Is this comment still needed?

Reply:
A reasonable question :-P. I added some additional comments to the test file to explain the commented lines. They're just present for reference, since these parts of the test case should be truncated by the block function.